packages/core/src/document/editor.tsx

```diff
@@ -29,81 +35,46 @@ export const DocumentEditor: React.FunctionComponent<
 > = props => {
   const container = React.useRef<HTMLDivElement>(null)
 
-  const identifier = props.state
+  const identifier = props.state || createDocument()
   const { id } = identifier
 
   const store = React.useContext(EditorContext)
 
-  React.useEffect(() => {
-    if (!getDocument(store.state, id)) {
+  const deserializeHelpers: StoreDeserializeHelpers = {
+    createDocument(document: {
+      id: string
+      plugin?: string
+      state?: unknown
+    }): void {
       const plugin = getPlugin(
         store.state,
-        identifier.plugin || store.state.defaultPlugin
+        document.plugin || getDefaultPlugin(store.state)
       )
-
-      if (plugin && isStatefulPlugin(plugin)) {
-        const state = plugin.state(
-          deserializeDocument(identifier).state,
-          undefined,
-          () => {}
-        )
-
-        store.dispatch({
-          type: ActionType.Insert,
-          payload: {
-            ...identifier,
-            plugin: identifier.plugin || store.state.defaultPlugin,
-            state: state.$$value
-          }
-        })
-      } else {
-        store.dispatch({
-          type: ActionType.Insert,
-          payload: {
-            ...identifier,
-            plugin: identifier.plugin || store.state.defaultPlugin
-          }
-        })
-      }
       store.dispatch({
-        type: ActionType.ResetHistory
+        type: ActionType.Insert,
+        payload: {
+          id: document.id,
+          plugin: document.plugin || getDefaultPlugin(store.state),
+          ...(plugin && isStatefulPlugin<StateDescriptor<any>>(plugin)
+            ? {
+                state: document.state
```

Comment:
Check for `undefined` instead. Otherwise, we will have problems with plugins that have a falsy state (e.g. a plugin with `StateType.boolean` and value `false`).